cw-orch/src/snapshots.rs

```diff
@@ -1,30 +1,41 @@
 //! Defined the snapshot testing macro on the Mock environment
 //! This is included here and not in the mock package because it needs to import traits to work
 
+/// Function helper used to parse storage into readable strings
+pub fn parse_storage(storage: Vec<(Vec<u8>, Vec<u8>)>) -> Vec<(String, String)> {
+    storage
+        .iter()
+        .map(|(key, value)| {
+            (
+                String::from_utf8_lossy(key).to_string(),
+                String::from_utf8_lossy(value).to_string(),
+            )
+        })
+        .collect::<Vec<_>>()
+}
 /// This functions allows for making sure the tests are stabilised and changes made to contracts don't have an impact on the internal storage
 /// This should ONLY be used when cw_orch is in scope
 /// Usage:
 /// ```rust,ignore
-/// mock.take_storage_snapshot("mock_doc");
+/// take_storage_snapshot!(chain, "mock_doc");
 /// ```
 /// The name you input to the function should be different from all other snapshots in your repository
 /// Find more details on how snapshot testing works on the official quick-start guide: https://insta.rs/docs/quickstart/
 /// This function will panic if the snapshot is different from the reference snapshot
+
 #[macro_export]
 macro_rules! take_storage_snapshot {
     ($chain: ident, $name: literal) => {
         // We register and test a snapshot for all contracts storage
         use ::cw_orch::environment::{ChainState as _, StateInterface as _};
         let all_contract_addresses = $chain.state().get_all_addresses()?;
-        for (_id, contract_addr) in all_contract_addresses {
+        for (id, contract_addr) in all_contract_addresses {
             ::cw_orch::insta::assert_yaml_snapshot!(
```

Comment:
Is it possible to have a single "assert snapshot" after loop? Right now I have to re-run snapshot test for every contract address